base/range.jl

```diff
@@ -1004,19 +1008,9 @@ function getindex(r::StepRange, s::AbstractRange{T}) where {T<:Integer}
     @boundscheck checkbounds(r, s)
 
     if T === Bool
-        if length(s) == 0
-            start, len = first(r), 0
-        elseif length(s) == 1
-            if first(s)
-                start, len = first(r), 1
-            else
-                start, len = first(r), 0
-            end
-        else # length(s) == 2
-            start, len = last(r), 1
-        end
-        return range(start, step=step(r); length=len)
+        range(first(s) ? first(r) : last(r), step = step(r), length = Int(last(s)))
```

Comment:
This branch seems broken if `s` isa `StepRangeLen{Bool}`
```julia
julia> (1:2:3)[StepRangeLen{Bool}(true,-1,2)]
1:2:0  #3:2:3 on master (also not correct but at least not empty.)

julia> (1:2:3)[StepRangeLen{Bool}(true,-1,2) |> collect]
1-element Vector{Int64}:
 1
```
Since this change doesn't help to fix #40760, should we keep the version on master?